daffodil-lib/src/test/scala/org/apache/daffodil/validation/TestValidatorsSPI.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.validation
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertThrows
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class TestValidatorsSPI {
+  val schema = getClass.getResource("/test/validation/testSchema1.dfdl.xsd").toURI.toString
+  val infoset = getClass.getResourceAsStream("/test/validation/testData1Infoset.xml")
+
+  @Test def testValidatorGetNotFoundThrows(): Unit = {
+    assertThrows(classOf[ValidatorNotRegisteredException], () => Validators.get("dont exist"))
```

Comment:
Note/FYI: we also have an Intercept implicit that can be used to do this sort of checking. Search tests for "intercept[".
Honestly I did not even know of assertThrows until now, but it makes sense that it would be in junit.